### PR TITLE
Android/Fix: Read User Authentication token from the environment variable

### LINKED ISSUE #3795: Read sentry token from environment variable (SENTRY_AUTH_TOKEN)
## Current behaviour

The android script only reads the auth token from `sentry.properties`
However this file is usually checked into git, making the auth key vulnerable.

https://github.com/getsentry/sentry-react-native/blob/7335e6ef8c0eb8aea2418e15feb770818bc4e587/sentry.gradle#L186

Error logs : 
```
Execution failed for task ':app:createBundleInternalReleaseJsAndAssets_SentryUpload_com.xxx.android@6.31.0-internal+1157_1157'.
> Illegal null value provided in this collection: [--log-level, debug, --url, https://sentry-dev.xxx.com/, --auth-token, null, react-native, gradle, --bundle, /Users/vagrant/git/android/app/build/generated/assets/createBundleInternalReleaseJsAndAssets/index.android.bundle, --sourcemap, /Users/vagrant/git/android/app/build/generated/sourcemaps/react/internalRelease/index.android.bundle.map, --org, xxx, --project, xxx, --release, com.xxx.android@6.31.0-internal+1157, --dist, 1157]
```

## Expected

Android and iOS source upload scripts must respect the `SENTRY_AUTH_TOKEN` if present

This enables us to keep the auth token as a secret in our CI environments.

---

note: it looks like iOS does read from the environment variable, but I couldn't confirm it in source code. It would be helpful if the maintainers could point out where this happens

related issue : https://github.com/getsentry/sentry-react-native/issues/1888 (But does not work)

### PR DESCRIPTION
**Pull request by lucas-zimerman**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

In short, use the authentication token if defined on sentry.properties, fallback to the environment variable if defined, otherwise build will fail.

## :bulb: Motivation and Context

To fix #3795.

## :green_heart: How did you test it?
 locally:
export SENTRY_AUTH_TOKEN="ABC"  then `./gradlew assembleRelease`
image
export SENTRY_AUTH_TOKEN="VALID TOKEN"
image

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [ ] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

Closes #3795

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 ### Fixes
 
 - `sentry-expo-upload-sourcemaps` no longer requires Sentry url when uploading sourcemaps to `sentry.io` ([#3915](https://github.com/getsentry/sentry-react-native/pull/3915))
+- Android projects can now use the environment variable `SENTRY_AUTH_TOKEN` during the release build. ([#3917](https://github.com/getsentry/sentry-react-native/pull/3917))
```
Comment (krystofwoldrich) [inline CHANGELOG.md:12]: Just making the change log more accurate about what situation is being fixed. 
```suggestion
- Flavor aware Android builds use `SENTRY_AUTH_TOKEN` env as fallback when token not found in `sentry-flavor-type.properties`. ([#3917](https://github.com/getsentry/sentry-react-native/pull/3917))
```

**Comment by krystofwoldrich** [comment]
@lucas-zimerman Have you tested this with the `flavorAware` flag enabled?

From the PR description, I'm not sure.

**Comment by lucas-zimerman** [comment]
> @lucas-zimerman Have you tested this with the `flavorAware` flag enabled?
> 
> From the PR description, I'm not sure.

I have tested with the default value so FlavorAware was off.

I have tried with flavor aware = true but it seems like the same isn't working for me (build error) but the reasons seems unrelated to this PR.

**Comment by lucas-zimerman** [comment]
@krystofwoldrich after finding the correct flavoraware filename, I was able to test that there are no additional changes for it to read from the environment file:
image

**Review by krystofwoldrich — APPROVED** [review]
Thank you! 🚀

**Pull request merged**